### PR TITLE
[ENHANCEMENT+BREAKING CHANGE] Migrate DataGrid to v7

### PR DESCRIPTION
**Pull request by celian-garcia**
> [!CAUTION]
> Don´t squash as we need the two commits in master history
# Description

I realized that columns are resizable in the MIT free version of datagrid in v7 (was not free in v6)
So I'm following this documentation to migrate
https://mui.com/x/migration/migration-data-grid-v6/

This will be helpful if we choose DataGrid for table panels for example

A consequence of that is that we need to update mui/material and the ``TimeRangeSelector`` was not working anymore properly (the select was listening for new value even if MenuItem was not containing value). That's why I'm refactoring the ``TimeRangeSelector`` in the same PR. 

# Screenshots

No changes

# Checklist

- [x] Pull request has a descriptive title and context useful to a reviewer.
- [x] Pull request title follows the `[<catalog_entry>] <commit message>` naming convention using one of the
  following `catalog_entry` values: `FEATURE`, `ENHANCEMENT`, `BUGFIX`, `BREAKINGCHANGE`, `DOC`,`IGNORE`.
- [x] All commits have [DCO signoffs](https://github.com/probot/dco#how-it-works).

## UI Changes

- [x] Changes that impact the UI include screenshots and/or screencasts of the relevant changes.
- [x] Code follows the [UI guidelines](https://github.com/perses/perses/blob/main/ui/ui-guidelines.md).
- [x] Visual tests are stable and unlikely to be flaky.
  See [Storybook](https://github.com/perses/perses/tree/main/ui/storybook#visual-tests)
  and [e2e](https://github.com/perses/perses/tree/main/ui/e2e#visual-tests) docs for more details. Common issues
  include:
  - Is the data inconsistent? You need to mock API requests.
  - Does the time change? You need to use consistent time values or mock time utilities.
  - Does it have loading states? You need to wait for loading to complete.

### CODE REVIEW

**Comment by celian-garcia** [comment]
I don´t get why the custom "designSystem" palette is not recognized anymore in some places only. Despite the typescript augment that I didn't modify.

I didn't find anything related to that in the release notes. The doc still advise to do it that way. Still investigating...

File: ui/e2e/src/fixtures/dashboardTest.ts [hunk 1]
```diff
@@ -72,7 +72,7 @@ async function duplicateDashboard(projectName: string, dashboardName: string, ne
   const result = await createDashboard(newDashboardJson);
   if (!result.ok) {
     throw new Error(
-      `Unable to create test dashboard '${newDashboardName}'. Failed with status '${result.status}: ${result.statusText}'.`
+      `Unable to create test dashboard '${newDashboardName}'. Failed with status '${result.status}: ${result.statusText}'. \n${await result.text()}`
```
Comment (Gladorme) [inline ui/e2e/src/fixtures/dashboardTest.ts:75]: Really useful?

Comment (celian-garcia) [reply ui/e2e/src/fixtures/dashboardTest.ts:75]: Yes for the tests, when we see a 400 we can know why we see a 400 this is a game changer!

File: ui/components/src/TimeRangeSelector/TimeRangeSelector.tsx [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+        open={showCustomDateSelector}
+        onClose={() => setShowCustomDateSelector(false)}
+        sx={(theme) => ({
+          padding: theme.spacing(2),
+        })}
+      >
+        <DateTimeRangePicker
+          initialTimeRange={convertedTimeRange}
+          onChange={(value: AbsoluteTimeRange) => {
+            // setLastOption({ value: value, display: formatTimeRange(value, timeZone) });
```
[the remark this answers is not in the document]
Comment (celian-garcia) [reply ui/components/src/TimeRangeSelector/TimeRangeSelector.tsx:96]: !

File: ui/components/src/TimeRangeSelector/TimeRangeSelector.tsx [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+              key={idx}
+              value={formatTimeRange(item.value, timeZone)}
+              onClick={() => {
+                onChange(item.value);
+              }}
+            >
+              {item.display}
+            </MenuItem>
+          ))}
+          {/*TODO(cegarcia): Is that ok to call a function here formatTimeRange*/}
```
[the remark this answers is not in the document]
Comment (celian-garcia) [reply ui/components/src/TimeRangeSelector/TimeRangeSelector.tsx:137]: !

**Review by Gladorme — APPROVED** [review]

**Pull request merged**